### PR TITLE
feat(bridge): RX-ify the dashboard component

### PR DESCRIPTION
**Pull request by heinzburgstaller**
## This PR
To introduce state management the dashboard component has to be refactored to use dumb components.
- Refactor dashboard to be a fully smart component, which fetches all relevant data for its dumb child components
- Dumbify project list, project tile and sequences state list

### Related Issues
Adds to #8023 

### Notes
This is phase 1 of the refactoring plan:
IMG_20220622_163331

Signed-off-by: Heinz Burgstaller <heinz.burgstaller@dynatrace.com>

### CODE REVIEW

File: bridge/client/app/_components/ktb-project-list/ktb-project-list.component.ts [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
+    const latestSequences = this.sequences[project.projectName];
+    return latestSequences ?? [];
   }
 
-  set projects(value: Project[]) {
-    if (this._projects !== value) {
-      this._projects = value;
-    }
+  getShipyardversion(): string | undefined {
+    return this.metadata != null ? this.metadata?.shipyardversion : undefined;
```
Comment (Kirdock) [inline bridge/client/app/_components/ktb-project-list/ktb-project-list.component.ts:25]: Can be simplified to `return this.metadata?.shipyardversion`. If `metadata` is `null`, `metadata?.shipyardversion` will be `undefined`. In my opinion we don't need an extra function for that and simply use `metadata?.shipyardversion` in the template

File: bridge/client/app/dashboard-legacy/dashboard-legacy.component.spec.ts [hunk 2]
```diff
@@ -12,7 +13,7 @@ describe('DashboardLegacyComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [HttpClientTestingModule],
-      providers: [],
+      providers: [{ provide: POLLING_INTERVAL_MILLIS, useValue: 0 }],
```
Comment (Kirdock) [inline bridge/client/app/dashboard-legacy/dashboard-legacy.component.spec.ts:16]: We don't need this provider as we call the constructor of this component on our own

**Review by Kirdock — CHANGES_REQUESTED** [review]

File: bridge/client/app/dashboard-legacy/dashboard-legacy.component.spec.ts (hunk 2 above)
Comment (heinzburgstaller) [reply bridge/client/app/dashboard-legacy/dashboard-legacy.component.spec.ts]: Fixed

File: bridge/client/app/_components/ktb-sequence-state-list/ktb-sequence-state-list.component.ts [hunk 3]
```diff
@@ -14,31 +11,15 @@ import { Router } from '@angular/router';
   styleUrls: [],
   encapsulation: ViewEncapsulation.None,
   preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class KtbSequenceStateListComponent implements OnDestroy {
-  @HostBinding('class') cls = 'ktb-sequence-state-list';
-  private _project?: Project;
+export class KtbSequenceStateListComponent {
+  @Input() project?: Project;
```
[the remark this answers is not in the document]
Comment (heinzburgstaller) [reply bridge/client/app/_components/ktb-sequence-state-list/ktb-sequence-state-list.component.ts]: Fixed

File: bridge/client/app/dashboard-legacy/dashboard-legacy.component.spec.ts [hunk 4]
```diff
@@ -49,7 +51,26 @@ describe('DashboardLegacyComponent', () => {
     expect(loadProjectSpy).toHaveBeenCalled();
   });
 
+  it('should load a maximum of 5 sequences per project', (done) => {
```
Comment (heinzburgstaller) [inline bridge/client/app/dashboard-legacy/dashboard-legacy.component.spec.ts:54]: @renepanzar @Kirdock @ermin-muratovic do you have a better solution to test complex RXJS observables?

**Review by Kirdock — APPROVED** [review]

**Pull request merged**